nexus_keycode/protocols/small.py

```diff
@@ -251,15 +253,18 @@ def __init__(self, id_, days, secret_key):
 
     @classmethod
     def generate_body(cls, days):
-        if 1 <= days <= 180:
-            increment_id = days - 1
-        elif 181 <= days <= cls.MAX_ADD_CREDIT_DAYS:
-            increment_id = ((days - 181) / cls.COARSE_DAYS_PER_INCREMENT_ID) + 180
+        if isinstance(days, int):
+            if 1 <= days <= 180:
+                increment_id = days - 1
+            elif 181 <= days <= cls.MAX_ADD_CREDIT_DAYS:
+                increment_id = ((days - 181) // cls.COARSE_DAYS_PER_INCREMENT_ID) + 180
```

Comment:
Hmm, this might hide errors that come from trying to use an unsupported/invalid number of days.

Explicitly, smallpad **does not** support single-day increments above 180, so I'm afraid that the flooring logic will 'hide' problems (user asks for 188 days, we don't support that in smallpad. In the code before this commit, that would throw an error (eventually?, I think?). In this change, it will silently give them a number of days not equal to 188.

Reply:
Maybe another solution could be to check for integer divisbility of `(days - 181)` by `cls.COARSE_DAYS_PER_INCREMENT_ID` and raise an error if its not divisible.